RecoLocalMuon/GEMSegment/plugins/ME0SegAlgoRU.cc

```diff
@@ -0,0 +1,456 @@
+	stdParameters.maxChi2Additional = ps.getParameter<double>("maxChi2Additional");
+	stdParameters.maxChi2Prune      = ps.getParameter<double>("maxChi2Prune"     );
+	stdParameters.maxChi2GoodSeg    = ps.getParameter<double>("maxChi2GoodSeg"   ) ;
+	stdParameters.maxPhiSeeds       = ps.getParameter<double>("maxPhiSeeds"      ) ;
+	stdParameters.maxPhiAdditional  = ps.getParameter<double>("maxPhiAdditional" );
+	stdParameters.maxETASeeds       = ps.getParameter<double>("maxETASeeds"      );
+	stdParameters.maxTOFDiff        = ps.getParameter<double>("maxTOFDiff"      );
+	stdParameters.minNumberOfHits        = ps.getParameter<unsigned int>("minNumberOfHits"      );
+	stdParameters.requireBeamConstr = true;
+
+	wideParameters = stdParameters;
+	wideParameters.maxChi2Prune     *=2;
+	wideParameters.maxChi2GoodSeg   *=2;
+	wideParameters.maxPhiSeeds      *=2;
+	wideParameters.maxPhiAdditional *=2;
+	wideParameters.minNumberOfHits  +=1;
+
+
+	displacedParameters = stdParameters;
+	displacedParameters.maxChi2Additional *= 2;
+	displacedParameters.maxChi2Prune      = 100;
+	displacedParameters.maxChi2GoodSeg    *= 5;
+	displacedParameters.maxPhiSeeds       *=2;
+	displacedParameters.maxPhiAdditional  *=2;
+	displacedParameters.maxETASeeds       *=2;
+	displacedParameters.requireBeamConstr = false;
+
+
+	LogDebug("ME0SegAlgoRU") << myName << " has algorithm cuts set to: \n"
+			<< "--------------------------------------------------------------------\n"
+			<< "allowWideSegments   = " <<allowWideSegments              << "\n"
+			<< "doCollisions        = " <<doCollisions                   << "\n"
+			<< "maxChi2Additional   = " <<stdParameters.maxChi2Additional<< "\n"
+			<< "maxChi2Prune        = " <<stdParameters.maxChi2Prune     << "\n"
+			<< "maxChi2GoodSeg      = " <<stdParameters.maxChi2GoodSeg   << "\n"
+			<< "maxPhiSeeds         = " <<stdParameters.maxPhiSeeds      << "\n"
+			<< "maxPhiAdditional    = " <<stdParameters.maxPhiAdditional << "\n"
+			<< "maxETASeeds         = " <<stdParameters.maxETASeeds      << "\n"
+			<< "maxTOFDiff          = " <<stdParameters.maxTOFDiff       << "\n"
+			<< std::endl;
+
+	theChamber=0;
+
+}
+
+std::vector<ME0Segment> ME0SegAlgoRU::run(const ME0Chamber * chamber, const HitAndPositionContainer& rechits){
+
+#ifdef EDM_ML_DEBUG // have lines below only compiled when in debug mode
+	ME0DetId chId(chamber->id());
+	edm::LogVerbatim("ME0SegAlgoRU") << "[ME0SegmentAlgorithm::run] build segments in chamber " << chId << " which contains "<<rechits.size()<<" rechits";
+	for(unsigned int iH = 0; iH < rechits.size(); ++iH){
+		auto me0id = rechits[iH].rh->me0Id();
+		auto rhLP  = rechits[iH].lp;
+		edm::LogVerbatim("ME0SegAlgoRU") << "[RecHit :: Loc x = "<<std::showpos<<std::setw(9)<<rhLP.x()<<" Glb y = "<<std::showpos<<std::setw(9)<<rhLP.y()<<" Time = "<<std::showpos<<rechits[iH].rh->tof()<<" -- "<<me0id.rawId()<<" = "<<me0id<<" ]"<<std::endl;
+	}
+#endif
+
+
+	if (rechits.size() < 3 || rechits.size()>300){

```

Comment:
It is sufficient for the current simulation with PU 200. We may want to make it a configuration parameter in the case that something drastic changes (e.g. large increase in neutron bkg or noise).